modules/storages/lib/api/v3/storages/storage_representer.rb

```diff
@@ -178,12 +178,14 @@ def initialize(model, current_user:, embed_links: nil)
     end
 
     associated_resource :oauth_application,
-                        skip_render: ->(*) { !current_user.admin? },
+                        skip_render: ->(*) { !current_user.admin? || !represented.provider_type_nextcloud? },
                         getter: ->(*) {
+                          next unless current_user.admin? && represented.provider_type_nextcloud?
+
                           ::API::V3::OAuth::OAuthApplicationsRepresenter.create(represented.oauth_application, current_user:)
                         },
                         link: ->(*) {
-                          next unless current_user.admin?
+                          next unless current_user.admin? && represented.provider_type_nextcloud?
```

Comment:
🟡 extract this bool condition to a meaningful method name and use it at all three places (first one after applying De Morgan's rule).

e.g.

```ruby
def represent_oauth_application?
  current_user.admin? && represented.provider_type_nextcloud?
end
```